test/1_stdlib/ErrorHandling.swift

```diff
@@ -340,4 +340,42 @@ ErrorHandlingTests.test("ErrorHandling/Collection map") {
   }
 }
 
+/* when b == 2 we are losing elements. insertion sort throws error
+in between while sorting so loop is not able to complete and swap the elements and array mutates
+to [2,3,3]
+ErrorHandlingTests.test("ErrorHandling/sort") {
+  var collection = [3, 2, 1]
+  do {
+    try collection.sort { (a, b) throws -> Bool in
+      print("collection before error \(collection)")
+      print("a \(a) \(b)")
+
+      if b == 2 {
```

Comment:
It will give error. I have tried that. Error: "A function declared rethrows may only throw if  parameter does". 
If I make some other function say `_insertionSort` to `throws` instead of `rethrows` than it will give error while calling `try _insertionSort(&elements, range , isOrderedBefore)` // Error: "Call can throw but error is nit handled; a function decalred rethrows may only throw if parameter does". It might be case we are calling `sorted` without throwing clousre.